File: src/main/java/com/rtomyj/skc/config/CORSConfig.java
package com.rtomyj.skc.config;

import org.jetbrains.annotations.NotNull;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;


@Configuration
public class CORSConfig
{
    @Bean
    public WebMvcConfigurer corsConfigurer() {
        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(@NotNull final CorsRegistry registry) {
                registry.addMapping("/**")
                        .allowedOrigins("http://localhost:3000", "https://thesupremekingscastle.com", "https://www.thesupremekingscastle.com", "https://dev.thesupremekingscastle.com")
                        .allowedMethods("GET")
                        .maxAge(21600); // 6 hours
            }
        };
    }
}

File: src/main/java/com/rtomyj/skc/config/CORSConfig.kt
package com.rtomyj.skc.config

import org.springframework.stereotype.Component
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Component
class CORSConfig: WebMvcConfigurer {
	override fun addCorsMappings(registry: CorsRegistry) {
		registry.addMapping("/**")
			.allowedOrigins(
				"http://localhost:3000",
				"https://thesupremekingscastle.com",
				"https://www.thesupremekingscastle.com",
				"https://dev.thesupremekingscastle.com"
			)
			.allowedMethods("GET")
			.maxAge(21600) // 6 hours
	}
}